packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -151,6 +153,7 @@ abstract class SerializationManager {
   /// Deserialize the provided json [data] by using the className stored in the [data].
   dynamic deserializeByClassName(Map<String, dynamic> data) {
     var className = data['className'];
+    if (className == 'null') return null;
     switch (className) {
```

Comment:
Chore: Same here, please connect this with the rest of the switch statement, no reason to have a different case for it.